packages/lodestar/src/db/api/beacon/stateContextCache.ts

```diff
@@ -26,12 +30,20 @@ export class StateContextCache {
     return this.clone(item);
   }
 
+  /**
+   * Add to both in-memory cache and state repository.
+   */
   public async add(item: ITreeStateContext): Promise<void> {
     this.cache[toHexString(item.state.hashTreeRoot())] = this.clone(item);
+    await this.stateRepo.add(item.state);
```

Comment:
Keep in mind storing 200mb state on disk after every slot takes. Maybe don't await it?